superchain/superchain-version_test.go

```diff
@@ -0,0 +1,61 @@
+package superchain_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func checkErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will fail if the superchain semver file
+// is not read correctly.
+func TestContractVersionsCheck(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+
+	desiredSemver := superchain.SuperchainSemver.OptimismPortal
+	// TODO complete all members of SuperchainSemver
+	// TODO resolve missing contracts
+	// TODO resolve extraneous contracts
```

Comment:
I notice there is not a 1-1 mapping between the set of contract names with declared desired semver, and the set of contract names for each chain. Also some chains have contracts which others lack.